### PR TITLE
feat(service-portal): Add bulk archive and fav

### PR DESCRIPTION
**Pull request by thordurhhh**
## What

Add bulk archive and favourite options to new inbox.
UI and service updates.

## Why

New functionality ✨ 

## Screenshots / Gifs

Screenshot 2023-10-11 at 10 17 57

Screenshot 2023-10-11 at 10 17 51

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/documents/src/lib/document.service.ts [hunk 1]
```diff
@@ -223,4 +224,32 @@ export class DocumentService {
...
+      )
+      return {
+        success: true,
+        messageId: stringIds,
+      }
+    } catch (e) {
+      logger.error('Post batch action failed', {
+        category: LOG_CATEGORY,
+        error: e,
+      })
```
Comment (thorkellmani) [inline libs/api/domains/documents/src/lib/document.service.ts:248]:
```suggestion
      logger.error('Post batch action failed', {
        category: LOG_CATEGORY,
        ...e,
      })
```
Do we want to log the action or the status to improve the debugging experience?

File: libs/service-portal/documents/codegen.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  libs/service-portal/documents/src/:
+    preset: 'near-operation-file'
+    presetConfig:
+      baseTypesPath: '~@island.is/api/schema'
+    plugins:
+      - typescript-operations
+      - typescript-react-apollo
+    config:
+      scalars:
+        DateTime: string
```
Comment (thorkellmani) [inline libs/service-portal/documents/codegen.yml:15]: I know we repeatedly convert the graphql datetimes to string in the config, but why?

File: libs/service-portal/documents/src/screens/Overview/Overview.tsx [hunk 3]
```diff
@@ -148,7 +169,7 @@ export const ServicePortalDocuments = () => {
     }
   }
 
-  const { data, totalCount, loading, error, refetch } = useListDocuments({
+  const { data, totalCount, loading, refetch } = useListDocuments({
```
Comment (thorkellmani) [inline libs/service-portal/documents/src/screens/Overview/Overview.tsx:172]: No error display if something goes wrong?

**Review by thorkellmani — CHANGES_REQUESTED** [review]
Logging!

File: libs/service-portal/documents/codegen.yml (hunk 2 above)
Comment (thordurhhh) [reply libs/service-portal/documents/codegen.yml:15]: I'm going to remove this.

**Review by thorkellmani — APPROVED** [review]
LGTM, might want to split the Overview up going forward. Bit busy

**Pull request merged**